meteor/rsmap.py

```diff
@@ -295,30 +309,86 @@ def canonicalize_amplitudes(self) -> None:
         )
 
     def to_structurefactor(self) -> rs.DataSeries:
+        """Return a DataSeries of complex structure factor amplitudes"""
         return super().to_structurefactor(self._amplitude_column, self._phase_column)
 
+    @overload
+    @classmethod
+    def from_structurefactor(
+        cls,
+        complex_structurefactor: rs.DataSeries,
+        *,
+        index: None = None,
+        cell: CellType | None = None,
+        spacegroup: SpacegroupType | None = None,
+    ) -> Map: ...
+
+    @overload
+    @classmethod
+    def from_structurefactor(
```

Comment:
maybe this name is a little too vague. Not clear that structurefactor is an `rs.DataSeries`.

Reply:
I tend to agree, but this is overwriting a method from `reciprocalspaceship`, so I would vote we keep it consistent.

https://github.com/rs-station/reciprocalspaceship/blob/cbf30c1c8438326baddf6a9898c2f5b5be3f7b43/reciprocalspaceship/dataset.py#L484-L528